xc/device.ml

```diff
@@ -1963,6 +1974,96 @@ module Dm = struct
         List.iter (write_local_domain "data/") ["updated"]
       | _ -> ()
 
+  (* Create two hash tables to store <domid, qmp connction> and <qmp connection, domid> *)
+  let domid_to_connection_tbl = Hashtbl.create 128
```

Comment:
I don't see why domid_to_connection is necessary -- when the thread wakes up, it will have a list of fds to transform to domids, not the other way round.